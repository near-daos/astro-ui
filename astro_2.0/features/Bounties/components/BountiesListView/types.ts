import { Bounty } from 'types/bounties';
import { ReactNode } from 'react';
import { ProposalVariant } from 'types/proposal';

export type SectionItem = {
  id: string;
  title: string;
  proposer: string;
  proposalId: string;
  bounty?: Bounty;
  link: {
    pathname: string;
    query: {
      dao: string;
      bounty?: string;
      proposal?: string;
    };
  };
  completeHandler?: (
    id: number,
    variant: ProposalVariant.ProposeDoneBounty
  ) => void;
  content: ReactNode;
};
