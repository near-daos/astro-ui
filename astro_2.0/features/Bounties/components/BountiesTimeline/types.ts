export type TimelineMilestoneType =
  | 'Proposal Created'
  | 'Bounty Created'
  | 'Claim'
  | 'Pending Approval'
  | 'Complete Claim'
  | 'Claim Deadline'
  | 'Complete Bounty';

export type TimelineMilestone = {
  type: TimelineMilestoneType;
  date: Date;
  tooltip: string;
  color?: string;
};

export type TimelineClaim = {
  id: string;
  milestones: TimelineMilestone[];
  title: string;
  color: string;
  minDate?: Date;
  maxDate?: Date;
};

export type TimelineGroup = {
  id: string;
  name: string;
  isOpen: boolean;
  milestones: TimelineMilestone[];
  claims: TimelineClaim[];
  minDate?: Date;
  maxDate?: Date;
};

export type TimelineGranularity = 'month' | 'day' | 'hour';
