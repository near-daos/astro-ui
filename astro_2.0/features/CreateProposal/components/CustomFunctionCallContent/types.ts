export enum FunctionCallType {
  Custom = 'Custom',
  BuyNFTfromMintbase = 'BuyNFTfromMintbase',
  TransferNFTfromMintbase = 'TransferNFTfromMintbase',
  MintNFTsfromMintbase = 'MintNFTsfromMintbase',
  BuyNFTfromParas = 'BuyNFTfromParas',
  SwapsOnRef = 'SwapsOnRef',
  VoteInAnotherDao = 'VoteInAnotherDao',
  RemoveUpgradeCode = 'RemoveUpgradeCode',
  CreateRoketoStream = 'CreateRoketoStream',
}
