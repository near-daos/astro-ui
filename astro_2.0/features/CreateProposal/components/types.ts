export type CreateTransferInput = {
  token: string;
  amount: number;
  details: string;
  externalUrl: string;
  target: string;
  gas: number;
};
