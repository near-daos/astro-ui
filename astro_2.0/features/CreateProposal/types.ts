import { DeadlineUnit } from 'types/bounties';

type ExternalLink = {
  id: string;
  url: string;
};

export interface LinksFormData {
  links: ExternalLink[];
  details: '';
  externalUrl: '';
}

export interface BondsAndDeadlinesData {
  createProposalBond: number;
  proposalExpireTime: number;
  claimBountyBond: number;
  unclaimBountyTime: number;
  details: string;
  externalUrl: string;
}

export type CreateBountyInput = {
  token: string;
  amount: number;
  details: string;
  externalUrl: string;
  slots: number;
  deadlineThreshold: number;
  deadlineUnit: DeadlineUnit;
};

export type Member = {
  name: string;
  value: number;
};

export type TokenDistributionGroup = {
  name: string;
  isCustom: boolean;
  groupTotal: string;
  members?: Member[];
};

export interface TokenDistributionInput {
  groups: TokenDistributionGroup[];
}

export interface CreateTokenInput {
  details: string;
  externalUrl: string;
  tokenName: string;
  totalSupply: string;
  tokenImage: '';
}

export type CreateProposalAction = 'createGovernanceToken';
