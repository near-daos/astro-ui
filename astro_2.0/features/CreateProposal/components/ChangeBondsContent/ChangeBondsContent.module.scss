@use 'styles/breakpoints';
@import 'styles/mixins';
@import 'styles/typography';

.root {
  display: flex;
  flex-direction: column;
}

.row {
  align-items: flex-start;
  display: flex;
  flex-direction: column;
  white-space: nowrap;

  &:not(:last-of-type) {
    margin-bottom: 24px;
  }

  .iconWrapper {
    background-color: var(--color-neutral-30);
    border-radius: 50%;
    height: 24px;
    margin-right: 8px;
    width: 24px;
  }
}

.inline {
  display: grid;
  grid-column-gap: 24px;
  grid-template-columns: 200px 1fr;
}

.title {
  @extend %title3;
  margin-bottom: 24px;
}

.inputWrapper {
  /* stylelint-disable */
  /* Chrome, Safari, Edge, Opera */
  input::-webkit-outer-spin-button,
  input::-webkit-inner-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }

  /* Firefox */
  input[type='number'] {
    -moz-appearance: textfield;
  }
  /* stylelint-enable */

  input {
    font-weight: 700;
    @include placeholder {
      color: var(--color-neutral-40);
      transition: opacity 0.2s linear;
    }
  }
}

.wide {
  flex: 1;
}

.narrow {
  max-width: 60px;
}

.select {
  flex: 1;
  max-width: fit-content;

  button {
    border: 0;
    overflow: hidden;
  }

  .balance {
    display: none;
    margin-left: auto;
  }

  .row {
    width: 100%;
  }
}

.select ul {
  min-width: 260px;
  .balance {
    display: flex;
  }
}
.symbol {
  @include ellipse-text;
  max-width: 60px;
}

@media screen and (max-width: #{breakpoints.$screen-sm}px) {
  .inline {
    display: grid;
    grid-row-gap: 24px;
    grid-template-columns: 1fr;
  }
}
