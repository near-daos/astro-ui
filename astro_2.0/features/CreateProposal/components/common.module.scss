@import 'styles/mixins';

// TODO refactor all proposals to use this styles
.input {
  font-weight: 700 !important;

  @include placeholder {
    color: var(--color-neutral-40) !important;
    transition: opacity 0.2s linear !important;
  }

  /* stylelint-disable */
  /* Chrome, Safari, Edge, Opera */
  &::-webkit-outer-spin-button,
  &::-webkit-inner-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }

  /* Firefox */
  &[type='number'] {
    -moz-appearance: textfield;
  }
  /* stylelint-enable */
}
