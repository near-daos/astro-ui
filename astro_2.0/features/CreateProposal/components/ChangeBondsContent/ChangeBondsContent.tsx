import React, { FC } from 'react';
import cn from 'classnames';
import Decimal from 'decimal.js';
import { useFormContext } from 'react-hook-form';

import { Input } from 'components/inputs/Input';
import { InputWrapper } from 'astro_2.0/features/CreateProposal/components/InputWrapper';

import { DAO } from 'types/dao';
import { YOKTO_NEAR } from 'services/sputnik/constants';

import { InfoBlockWidget } from 'astro_2.0/components/InfoBlockWidget';

import styles from './ChangeBondsContent.module.scss';

interface ChangeBondsContentProps {
  dao: DAO;
}

export const ChangeBondsContent: FC<ChangeBondsContentProps> = ({ dao }) => {
  const { register } = useFormContext();

  return (
    <div className={styles.root}>
      <div className={styles.row}>
        <div className={styles.title}>Proposals</div>
        <div className={styles.inline}>
          <InputWrapper
            fieldName="createProposalBond"
            label="Bonds to create proposals"
          >
            <Input
              className={cn(styles.inputWrapper, styles.narrow)}
              type="number"
              placeholder={new Decimal(dao.policy.proposalBond)
                .div(YOKTO_NEAR)
                .toString()}
              min={0}
              isBorderless
              size="block"
              {...register('createProposalBond')}
            />
          </InputWrapper>
          <InputWrapper
            fieldName="proposalExpireTime"
            label="Time before proposals expire"
          >
            <Input
              className={cn(styles.inputWrapper, styles.narrow)}
              type="number"
              min={1}
              placeholder={new Decimal(dao.policy.proposalPeriod)
                .div('3.6e12')
                .toString()}
              isBorderless
              size="block"
              {...register('proposalExpireTime')}
            />
          </InputWrapper>
        </div>
      </div>
      <div className={styles.row}>
        <div className={styles.title}>Bounties</div>
        <div className={styles.inline}>
          <InputWrapper
            fieldName="claimBountyBond"
            label="Bonds to claim bounty"
          >
            <Input
              className={cn(styles.inputWrapper, styles.narrow)}
              type="number"
              min={0}
              placeholder={new Decimal(dao.policy.bountyBond)
                .div(YOKTO_NEAR)
                .toString()}
              isBorderless
              size="block"
              {...register('claimBountyBond')}
            />
          </InputWrapper>

          <InputWrapper
            fieldName="unclaimBountyTime"
            label="Time to unclaim a bounty without penalty"
          >
            <Input
              className={cn(styles.inputWrapper, styles.narrow)}
              type="number"
              min={1}
              placeholder={new Decimal(dao.policy.bountyForgivenessPeriod)
                .div('3.6e12')
                .toString()}
              isBorderless
              size="block"
              {...register('unclaimBountyTime')}
            />
          </InputWrapper>
        </div>
      </div>

      <div className={styles.row}>
        <InfoBlockWidget label="Target" value={dao.id} valueFontSize="S" />
      </div>
    </div>
  );
};
