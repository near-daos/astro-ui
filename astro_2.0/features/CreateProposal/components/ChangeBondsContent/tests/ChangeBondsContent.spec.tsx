import { render } from 'jest/testUtils';

import { DAO } from 'types/dao';

import { ChangeBondsContent } from 'astro_2.0/features/CreateProposal/components/ChangeBondsContent';

const formContextMock = {
  register: () => 0,
  formState: {
    errors: {},
  },
};

jest.mock('react-hook-form', () => {
  return {
    useFormContext: jest.fn(() => formContextMock),
  };
});

jest.mock('next-i18next', () => ({
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => {
    return {
      t: (str: string): string => str,
    };
  },
}));

describe('ChangeBondsContent', () => {
  it('Should render component', () => {
    const dao = {
      id: '123',
      policy: {
        proposalBond: '1',
        proposalPeriod: '2',
        bountyBond: '3',
        bountyForgivenessPeriod: '4',
      },
    } as unknown as DAO;

    const { container } = render(<ChangeBondsContent dao={dao} />);

    expect(container).toMatchSnapshot();
  });
});
