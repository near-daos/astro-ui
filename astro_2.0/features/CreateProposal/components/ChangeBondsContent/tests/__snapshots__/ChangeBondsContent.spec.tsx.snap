// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangeBondsContent Should render component 1`] = `
<div>
  <div
    class="root"
  >
    <div
      class="row"
    >
      <div
        class="title"
      >
        proposalCard.proposals
      </div>
      <div
        class="inline"
      >
        <div
          class="root"
        >
          <div
            class="label"
          >
            <span>
              proposalCard.proposalBonds
            </span>
          </div>
          <label
            class="input sizeBlock borderless inputWrapper narrow"
            for="1"
          >
            <div
              class="inputHolder"
            >
              <input
                class=""
                id="1"
                min="0"
                placeholder="1e-24"
                style="text-align: left;"
                type="number"
              />
            </div>
          </label>
        </div>
        <div
          class="root"
        >
          <div
            class="label"
          >
            <span>
              proposalCard.proposalBondExpirationTime
            </span>
          </div>
          <label
            class="input sizeBlock borderless inputWrapper narrow"
            for="2"
          >
            <div
              class="inputHolder"
            >
              <input
                class=""
                id="2"
                min="1"
                placeholder="5.5555555555555555556e-13"
                style="text-align: left;"
                type="number"
              />
            </div>
          </label>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="title"
      >
        proposalCard.proposalBounties
      </div>
      <div
        class="inline"
      >
        <div
          class="root"
        >
          <div
            class="label"
          >
            <span>
              proposalCard.proposalClaimBounty
            </span>
          </div>
          <label
            class="input sizeBlock borderless inputWrapper narrow"
            for="3"
          >
            <div
              class="inputHolder"
            >
              <input
                class=""
                id="3"
                min="0"
                placeholder="3e-24"
                style="text-align: left;"
                type="number"
              />
            </div>
          </label>
        </div>
        <div
          class="root"
        >
          <div
            class="label"
          >
            <span>
              proposalCard.proposalBountyUnclaimTime
            </span>
          </div>
          <label
            class="input sizeBlock borderless inputWrapper narrow"
            for="4"
          >
            <div
              class="inputHolder"
            >
              <input
                class=""
                id="4"
                min="1"
                placeholder="1.1111111111111111111e-12"
                style="text-align: left;"
                type="number"
              />
            </div>
          </label>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="root"
      >
        <div
          class="block"
        >
          <div
            class="label"
          >
            proposalCard.proposalTarget
          </div>
        </div>
        <div
          class="block"
        >
          <div
            class="valueSmall"
          >
            123
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
